### PR TITLE
WIP: T-Coffee

### PR DESCRIPTION
**Pull request by anatskiy**

### CODE REVIEW

File: tools/tcoffee/tcoffee.xml [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+            <option value="muscle_msa">Muscle</option>
+            <option value="t_coffee_msa">T-Coffee</option>
+            <option value="pcma_msa">pcma</option>
+            <option value="kalign_msa">kalign</option>
+            <option value="amap_msa">amap</option>
+            <option value="proda_msa">proda</option>
+            <option value="prank_msa">prank</option>
+        </param>
+
+        <section name="out" title="Output Format" expanded="True">
```
Comment (bgruening) [inline tools/tcoffee/tcoffee.xml]: You can use here a `type="select" multiple="true"` like here: https://github.com/bgruening/galaxytools/blob/master/tools/antismash/antismash.xml#L180

File: tools/tcoffee/tcoffee.xml [hunk 2]
```diff
@@ -0,0 +1,186 @@
+<tool id="tcoffee" name="T-Coffee" version="0.1">
```
Comment (bgruening) [inline tools/tcoffee/tcoffee.xml]: id="t_coffee"

File: tools/tcoffee/tcoffee.xml [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+            $out.format_phylip
+            $out.format_pir_seq
+            $out.format_fasta_seq
+            $out.format_score_ascii
+            $out.format_score_html
+        #end if
+]]>
+    </command>
+    <inputs>
+        <param name="input" type="data" format="fasta" label="Source file" />
```
Comment (bgruening) [inline tools/tcoffee/tcoffee.xml]: label="sequences in FASTA format"

File: tools/tcoffee/tcoffee.xml [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+
+        <param name="method1" type="select" label="Pairwise Sequence Alignment Method">
```
Comment (bgruening) [inline tools/tcoffee/tcoffee.xml]: Do you know what default is? You can use a `type="select" optional="True"` and remove the Default option.

File: tools/tcoffee/.shed.yml [hunk 5]
```diff
@@ -0,0 +1,12 @@
+name: tcoffee
```
Comment (bgruening) [inline tools/tcoffee/.shed.yml]: t_coffee

File: tools/tcoffee/.shed.yml [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+owner: bgruening
+description: A collection of tools for Computing, Evaluating and Manipulating Multiple Alignments of DNA, RNA, Protein Sequences and Structures
+homepage_url: http://www.tcoffee.org/Projects/tcoffee/index.html
+long_description: T-Coffee is a multiple sequence alignment package. You can use T-Coffee to align sequences or to combine the output of your favorite alignment methods (Clustal, Mafft, Probcons, Muscle...) into one unique alignment.
+remote_repository_url: https://github.com/bgruening/galaxytools/tree/master/tools/tcoffee
+type: unrestricted
+categories:
+- Sequence Alignment
+- DNA
+- RNA
```
Comment (bgruening) [inline tools/tcoffee/.shed.yml]: You can only use categories from https://toolshed.g2.bx.psu.edu
* Sequence Analysis

**Comment by nsoranzo** [comment]
Are you aware of https://toolshed.g2.bx.psu.edu/view/ayllon/tcoffee/  ?

**Comment by anatskiy** [comment]
@nsoranzo, sure, I'm basing on the mentioned tool and enhancing it a bit.

**Comment by nsoranzo** [comment]
@anatskiy Have you checked with the original author if he is still maintaining it? If it's abandoned, maybe the https://github.com/galaxyproject/tools-iuc repository would be a better place for it, since IUC is usually willing to adopt abandoned tools. We are using it at TGAC, so I'd be also interested for example.

**Comment by bgruening** [comment]
@nsoranzo we have contacted the authors and never got a reply.

**Comment by nsoranzo** [comment]
@bgruening tools-iuc then?

**Comment by bgruening** [comment]
Yes, this is an incubator :)

File: tools/tcoffee/tcoffee.xml [hunk 7]
```diff
@@ -0,0 +1,174 @@
+<tool id="t_coffee" name="T-Coffee" version="0.1">
```
Comment (bgruening) [inline tools/tcoffee/tcoffee.xml]: Can you please add ` profile="16.04"` to the tool tag?
This will hopefully fix the stderr error.

**Comment by nsoranzo** [comment]
@bgruening I actually completely forgot this PR in the rush to finish our work for http://biorxiv.org/content/early/2016/12/23/096529 , sorry about that. Our new wrapper started from https://toolshed.g2.bx.psu.edu/view/ayllon/tcoffee/ with the following modifications:
- Remove the `tcoffee.pl` Perl script
- Add CIGAR as possible output format using a Python script to convert `fasta_aln` to CIGAR
- Add requirement
- Add `<version_command>`
- Quote datasets in `<command>`
- Add multithreading support with `-n_core` option
- Use a multiselect for the output formats
- Add tests
- Fix indentation

We also added the possible to specify a second input file with a list of sequence IDs, which essentially integrates `filter_by_fasta_ids` into the `t_coffee` wrapper. I don't really like this, but we need it to reduce the number of steps in our workflow, which run on very large collections, otherwise the huge number of jobs kills our Galaxy job handlers.

To conclude, I think that we re-did most of things in this PR, except we are still using the t_coffee methods from the original wrapper, while here there seems to be more. If you want to contribute anything to our wrapper, PRs are obvoiusly welcome.

**Comment by bgruening** [comment]
Crashing the job handlers? How big are your worklfows, we are running here workflows with multiple 100 steps, nearly 1000?

So this means we will close this. @anatskiy sorry!
Will this wrapper enter IUC at some point?

**Comment by nsoranzo** [comment]
> - Add CIGAR as possible output format using a Python script to convert fasta_aln to CIGAR

FYI this has been removed from the wrapper, so it is now more similar to the original tool.

**Pull request closed without merging**